pyproject.toml

```diff
@@ -87,7 +87,7 @@ tests = [
     # version is pinned to the previous rose release)
 ]
 lint = [
-    "flake8>=4.0.0",
+    "flake8>=6.0.0; python_version > '3.7'",
```

Comment:
Flake8 v5 is the last version compatible with Python 3.7. However this version has a pin on `importlib-metadata` which conflicts with our own. So we can't run Flake8 on Python 3.7 anymore.